f3/granite.go

```diff
@@ -347,9 +361,9 @@ func (i *instance) beginQuality() {
 
 // Attempts to end the QUALITY phase and begin PREPARE based on current state.
 // No-op if the current phase is not QUALITY.
-func (i *instance) tryQuality() {
+func (i *instance) tryQuality() error {
 	if i.phase != QUALITY {
-		panic(fmt.Sprintf("unexpected phase %s", i.phase))
+		return fmt.Errorf("unexpected phase %s", i.phase)
```

Comment:
Note the expected phase, QUALITY, in the message (elsewhere too)

Reply:
✓